### PR TITLE
Fix --use-system-libraries not working with Rake install

### PR DESCRIPTION
**Pull request by stanhu**
Previously the libmagic tarball and patches were required to be present
in the current directory for `rake compile -- --use-system-libraries` to
work. To fix this, we now dynamically add these dependencies when
building the gem.

### CODE REVIEW

**Comment by stanhu** [comment]
@kwilczynski I think the build failures here are unrelated, so I'm just going to merge. 😄

**Comment by stanhu** [comment]
I guess I can't merge without approval. 😄

**Comment by kwilczynski** [comment]
Hi @stanhu,

> I guess I can't merge without approval. 😄

Sorry about that! Although, I am not sure why - the only thing I enabled (or re-enabled, rather) yesterday was requirement to sign commits, as per:

Screenshot 2021-04-20 at 08 14 04

Does it still pose problems?

Krzysztof

**Review by kwilczynski — APPROVED** [review]
Hi @stanhu, thank you for fixing this!

File: Rakefile
Comment (stanhu) [inline Rakefile]: Unintended comment.

**Comment by flavorjones** [comment]
Apologies for not sending a PR for this earlier! Thanks for taking care of it!

**Comment by kwilczynski** [comment]
Hi @flavorjones,

> Apologies for not sending a PR for this earlier! Thanks for taking care of it!

Thank **_YOU_** for doing all the work on enabling us to ship native Ruby Gems and to better the users experience!

Krzysztof

**Comment by stanhu** [comment]
Yes, thank you @flavorjones for helping us with this project! I cribbed more Nokogiri work into this project. 😄

**Pull request merged**
